src/client/pythonEnvironments/info/interpreter.ts

```diff
@@ -76,5 +78,42 @@ export async function getInterpreterInfo(
 }
 
 export function getInterpreterHash(interpreter: PythonEnvironment | {path: string}){
-    return sha256().update(interpreter.path).digest('hex');
+    const interpreterPath = getNormalizedInterpreterPath(interpreter.path);
+    return sha256().update(interpreterPath).digest('hex');
+}
+
+/**
+ * Sometimes on CI, we have paths such as (this could happen on user machines as well)
+ *  - /opt/hostedtoolcache/Python/3.8.11/x64/python
+ *  - /opt/hostedtoolcache/Python/3.8.11/x64/bin/python
+ *  They are both the same.
+ * This function will take that into account.
+ */
+export function areInterpreterPathsSame(path1: string = '', path2:string = '', ostype = getOSType(), fs?: IFileSystem){
```

Comment:
Added new helpers to compare paths, ensured the hash also takes this into account.